src/qcodes/instrument_drivers/CopperMountain/_M5xxx.py

```diff
@@ -0,0 +1,898 @@
+            "off if one wants to minimize overhead.",
+        )
+
+        # Electrical distance default units.
+        self.electrical_distance_units("MET")
+
+        self.connect_message()
+
+    def reset(self) -> None:
+        self.write("*RST")
+
+    def _set_start(self, val: float) -> None:
+        """Sets the start frequency and updates linear trace parameters.
+
+        Args:
+            val: start frequency to be set
+
+        Raises:
+            ValueError: If start > stop
+
+        """
+        stop = self.stop()
+        if val >= stop:
+            raise ValueError("Stop frequency must be larger than start frequency.")
+        self.write(f"SENS1:FREQ:STAR {val}")
+        # we get start as the vna may not be able to set it to the
+        # exact value provided.
+        start = self.start()
+        if abs(val - start) >= 1:
+            log.warning(f"Could not set start to {val} setting it to {start}")
+        self.update_lin_traces()
+
+    def _set_stop(self, val: float) -> None:
+        """Sets the start frequency and updates linear trace parameters.
+
+        Args:
+            val: start frequency to be set
+
+        Raises:
+            ValueError: If stop < start
+
+        """
+        start = self.start()
+        if val <= start:
+            raise ValueError("Stop frequency must be larger than start frequency.")
+        self.write(f"SENS1:FREQ:STOP {val}")
+        # We get stop as the vna may not be able to set it to the
+        # exact value provided.
+        stop = self.stop()
+        if abs(val - stop) >= 1:
+            log.warning(f"Could not set stop to {val} setting it to {stop}")
+        self.update_lin_traces()
+
+    def _set_span(self, val: float) -> None:
+        """Sets frequency span and updates linear trace parameters.
+
+        Args:
+            val: frequency span to be set
+
+        """
+        self.write(f"SENS1:FREQ:SPAN {val}")
+        self.update_lin_traces()
+
+    def _set_center(self, val: float) -> None:
+        """Sets center frequency and updates linear trace parameters.
+
+        Args:
+            val: center frequency to be set
+
+        """
+        self.write(f"SENS1:FREQ:CENT {val}")
+        self.update_lin_traces()
+
+    def _set_number_of_points(self, val: int) -> None:
+        """Sets number of points and updates linear trace parameters.
+
+        Args:
+            val: number of points to be set.
+
+        """
+        self.write(f"SENS1:SWE:POIN {val}")
+        self.update_lin_traces()
+
+    def _get_trigger(self) -> str:
+        """Gets trigger source.
+
+        Returns:
+            str: Trigger source.
+
+        """
+        r = self.ask("TRIG:SOUR?")
+
+        if r.lower() == "int":
+            return "internal"
+        elif r.lower() == "ext":
+            return "external"
+        elif r.lower() == "man":
+            return "manual"
+        else:
+            return "bus"
+
+    def _set_trigger(self, trigger: str) -> None:
```

Comment:
Nitpick:
Since the allowed strs come from a specific set, consider typing `trigger` as `Literal["external", "internal", "manual", "bus"]` This isn't required, as the Parameter Validator handles it, but it's nice.